### PR TITLE
Add check that all pages render

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Closes https://github.com/Qiskit/documentation/issues/511. Example:

```
$ npm run check-pages-render

Checked 10 / 71 pages
Checked 20 / 71 pages
Checked 30 / 71 pages
Checked 40 / 71 pages
Checked 50 / 71 pages
Checked 60 / 71 pages
Checked 70 / 71 pages
✅ All pages render without crashing
```

## Only checks non-API docs by default

This script is quite slow, at least on my M1 because the Docker images is built with x86. 

So, to avoid CI slowing down too much, we only check non-API pages in PR builds. Our nightly cron job checks everything else.

## Does not auto-start Docker

We no longer have the file `webServer.ts` thanks to https://github.com/Qiskit/documentation/pull/578, which was a great improvement. 

Rather than adding back somewhat complex code for us to auto-start the server—and then to periodically ping if it's ready or time out—we expect the user to start up the server. That's acceptable since usually people will rely on CI to run this check. It's too slow for people to be frequently running locally.

### CODE REVIEW

File: .github/workflows/nightly-extended-checks.yml [hunk 1]
```diff
@@ -33,3 +33,15 @@ jobs:
...
+        run: |
+          ./start &
+          sleep 20
+      - name: Check API pages render
+        run: |
+          npm run check-pages-render -- \
+            --qiskit-release-noets \
+            --current-apis \
+            --historical-apis \
+            --translations
```
Comment (frankharkins) [inline .github/workflows/nightly-extended-checks.yml:47]: Aesthetic nit but I recently learnt `>` removes newlines

```suggestion
        run: >
          npm run check-pages-render --
            --qiskit-release-noets
            --current-apis
            --historical-apis
            --translations
```

File: scripts/commands/checkPagesRender.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  }
+  if (!args.historicalApis) {
+    globs.push(
+      "!docs/api/{qiskit,qiskit-ibm-provider,qiskit-ibm-runtime}/[0-9]*/*",
+    );
+  }
+  if (!args.qiskitReleaseNotes) {
+    globs.push("!docs/api/qiskit/release-notes/*");
+  }
+  if (args.translations) {
```
Comment (frankharkins) [inline scripts/commands/checkPagesRender.ts:145]:
```suggestion
  if (!args.translations) {
```

File: .github/workflows/nightly-extended-checks.yml
[the remark this answers is not in the document]
Comment (Eric-Arellano) [reply .github/workflows/nightly-extended-checks.yml:1]: We don't ever have releases in this repo. One idea I was considering though is `branch` builds whenever API docs or translations change. Wdyt?

File: scripts/commands/checkPagesRender.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  }
+  if (!args.historicalApis) {
+    globs.push(
+      "!docs/api/{qiskit,qiskit-ibm-provider,qiskit-ibm-runtime}/[0-9]*/*",
+    );
+  }
+  if (!args.qiskitReleaseNotes) {
+    globs.push("!docs/api/qiskit/release-notes/*");
+  }
+  if (!args.translations) {
```
Comment (Eric-Arellano) [inline scripts/commands/checkPagesRender.ts:145]: Wait no, this was correct.

```suggestion
  if (!args.translations) {
```

File: .github/workflows/nightly-extended-checks.yml
[the remark this answers is not in the document]
Comment (frankharkins) [reply .github/workflows/nightly-extended-checks.yml:1]: That works! The case I'm imagining is where we merge a few changes rapidly then push them to production (which a nightly check might not catch in time)

File: scripts/commands/checkPagesRender.ts (hunk 3 above)
Comment (frankharkins) [reply scripts/commands/checkPagesRender.ts]: 🤦 sorry

**Review by arnaucasau — APPROVED** [review]
Looks good to me! Good idea with the `console.log` every 10 files!

**Pull request merged**